gems/smithy/lib/smithy/welds/endpoints.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Welds
+
+    # Provides default endpoint builtin/function bindings.
+    # TODO : There are many ways we could "register" the base (ie, non-AWS) endpoint builtins/functions
+    # I like this since it uses the same extension point that AWS will and makes it easy to remove/replace.
+    class Endpoints < Weld
+
+      def self.preprocess(model)
+        # TODO: There are quite a few places we could do this
+        # This is one idea - I like this because it keeps all of the "default" endpoint logic in an extension point rather than in core logic
+        # that makes it easy to remove/replace.
+        # add default endpoint rules/tests if none are set
+        model['shapes'].select { |_k, s| s['type'] == 'service' }.each do |_name, shape|
+          unless shape['traits']['smithy.rules#endpointRuleSet']
+            add_default_endpoints(shape['traits'])
+          end
+        end
+      end
+      def self.built_in_bindings
+        [Vise::Endpoints::BuiltInBinding.new(
+          id: 'SDK::Endpoint',
+          render_config: proc do |_plan|
+            <<-ADD_OPTION
+option(
+  :endpoint, 
+  doc_type: String,
+  docstring: "Custom Endpoint"
+)
+            ADD_OPTION
+          end,
+          render_build: proc do |_plan, _operation|
+            "config.endpoint"
+          end,
+          render_test_set: proc do |_plan, _operation, _node|
+
+          end
+        )]
+      end
+
+      def self.function_bindings
+        # TODO: Add all stdlib from:
+        # https://github.com/smithy-lang/smithy-ruby/blob/main/codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/rulesengine/FunctionBinding.java#L49
+        [Vise::Endpoints::FunctionBinding.new(
```

Comment:
I think these classes are basically views in anvil are they not? Vise would be for parsing the model

Reply:
What do you mean?  I'm not sure they really belong as views.   They're used on plan and I think thats probably where they belong (Since they relate to what is the broader generation context).